### DISCUSSION TITLE
Amazon aurora MySQL8.0互換のサポート

### DISCUSSION THREAD

**Discussion by sts-m-yokota** [domaframework/doma#907] [Q&A]
はじめまして。横田と申します。
新しいプロジェクトでKotlin + Spring boot + DOMA2 + Amazon aurora MySQL8.0互換　の採用を検討しています。
DOMA2の公式では特に明記はありませんが、導入可能なのか、または導入後に問題点が挙がっているのか、
現時点の情報を教えていただけますでしょうか。
不躾な質問となり申し訳ありませんが、どなたかご教授いただけますと幸いです。

**Comment by nakamura-to** [comment]
特にその組み合わせで動かないという情報はありません。どの辺りを気にされていますか？

ただ、Doma2をKotlinで使う場合には[kapt](https://kotlinlang.org/docs/kapt.html)を利用するのが一般的ですが、リンク先に記載があるようにkaptはメンテナンスモードになっていますので懸念点の1つと言えると思います。なお、DaoやEntityなどアノテーション処理の対象をKotlinではなくJavaで書くようにすればこの懸念点は回避できます。

Doma2の代わりに私がKotlin用に作っている[Komapper](https://github.com/komapper)を使うことも検討いただいても良いかもしれません。こちらはDoma2とアーキテクチャは異なりますが、2 Way SQLのような機能はありますしCriteria APIのような機能もあるという点では似ています。Doma2と同じくコンパイル時にアノテーション処理を行いますが、kaptではなく[KSP](https://github.com/google/ksp)というツールを使っています。注意点として、KomapperはDoma2に比べると実績や利用者はまだ少ないということが挙げられます。

**Reply by sts-m-yokota** [reply]
回答ありがとうございます！
Amazon aurora MySQL8.0互換とDOMA2を採用したとして、問題があった場合にORマッパーの再検討となることを懸念していました。
使い方次第で問題ないのかと思いました。
komapperのご提案もありがとうございます。
こちら参考にさせていただきます。
